buildinfo_test.go

```diff
@@ -230,6 +230,53 @@ func getFilesFromBuildDir(t *testing.T, buildName, buildNumber string) []os.File
 	return files
 }
 
+func TestBuildAppend(t *testing.T) {
+	initArtifactoryTest(t)
+	buildNumber1 := "12"
+	buildNumber2 := "13"
+
+	// Clean old builds tests if exists
+	inttestutils.DeleteBuild(artifactoryDetails.Url, tests.RtBuildName1, artHttpDetails)
+	inttestutils.DeleteBuild(artifactoryDetails.Url, tests.RtBuildName2, artHttpDetails)
+
+	// Publish build RtBuildName1/buildNumber1
+	err := artifactoryCli.WithoutCredentials().Exec("bce", tests.RtBuildName1, buildNumber1)
+	assert.NoError(t, err)
+	err = artifactoryCli.Exec("bp", tests.RtBuildName1, buildNumber1)
+	assert.NoError(t, err)
+
+	// Create a build RtBuildName2/buildNumber2 and append RtBuildName1/buildNumber1 to the build
+	err = artifactoryCli.Exec("ba", tests.RtBuildName2, buildNumber2, tests.RtBuildName1, buildNumber1)
+	assert.NoError(t, err)
+
+	// Assert RtBuildName2/buildNumber2 is appended to RtBuildName1/buildNumber1 locally
+	partials, err := utils.ReadPartialBuildInfoFiles(tests.RtBuildName2, buildNumber2)
+	assert.NoError(t, err)
+	assert.Len(t, partials, 1)
+	assert.Equal(t, tests.RtBuildName1+"/"+buildNumber1, partials[0].ModuleId)
+	assert.Equal(t, buildinfo.Build, partials[0].ModuleType)
+	assert.NotZero(t, partials[0].Timestamp)
+	assert.NotNil(t, partials[0].Checksum)
+	assert.NotZero(t, partials[0].Checksum.Md5)
+	assert.NotZero(t, partials[0].Checksum.Sha1)
+
+	// Publish build RtBuildName2/buildNumber2
+	err = artifactoryCli.Exec("bp", tests.RtBuildName2, buildNumber2)
+	assert.NoError(t, err)
+
+	// Check published build info
+	buildInfo, _, err := tests.GetBuildInfo(artifactoryDetails, tests.RtBuildName2, buildNumber2)
+	assert.NoError(t, err)
+	assert.NotNil(t, buildInfo)
+	assert.Len(t, buildInfo.BuildInfo.Modules, 1)
+	module := buildInfo.BuildInfo.Modules[0]
+	assert.Equal(t, tests.RtBuildName1+"/"+buildNumber1, module.Id)
+
+	// Clean builds
+	inttestutils.DeleteBuild(artifactoryDetails.Url, tests.RtBuildName1, artHttpDetails)
+	inttestutils.DeleteBuild(artifactoryDetails.Url, tests.RtBuildName2, artHttpDetails)
```

Comment:
I think we should call ```cleanArtifactoryTest()``` at the end fo this test. I actually noticed that there are a few other tests in this file which do not include this. Maybe we should add it to the end of all tests.